api4/apitestlib.go

```diff
@@ -120,6 +121,14 @@ func setupTestHelper(dbStore store.Store, searchEngine *searchengine.Broker, ent
 		options = append(options, app.StoreOverride(dbStore))
 	}
 
+	testLogger := mlog.NewLogger()
+	logCfg, _ := config.MloggerConfigFromLoggerConfig(memoryConfig.LogSettings, nil, config.GetLogFileLocation)
+	if errCfg := testLogger.ConfigureTargets(logCfg); errCfg != nil {
+		panic("failed to configure test logger: " + errCfg.Error())
+	}
+	testLogger.LockConfiguration()
```

Comment:
`LockConfiguration` is there because the logging engine setup by test helpers is blown away during server init, and again when certain services get enabled such as metrics collection for Logr. Previously there were cases where test helpers set a logging config that didn't actually stick during the tests. `LockConfiguration` allows us to force a config specified by a test helper to stay in effect throughout the server lifecycle.

It's true that `UnlockConfiguration` is never called. I put it there for completeness but can be removed.

Reply:
Ah I see. The semantics of the API confused me slightly. I think it might be better to pass an initial flag bit or something during initialization itself to indicate that this configuration cannot be changed. That way, we can avoid having a separate Lock method for this.

But anyways, this is also fine. Maybe a comment in the method which just explains what you wrote above would be helpful for future readers.

Reply:
Comments added.